okhttp/src/main/java/okhttp3/internal/http/HttpEngine.java

```diff
@@ -637,17 +581,14 @@ public void readResponse() throws IOException {
 
       httpStream.writeRequestHeaders(request);
 
-      //Update the networkRequest with the possibly updated interceptor request.
-      networkRequest = request;
```

Comment:
Apologies, I don't completely understand this. I think I see how the behavior is slightly different, but am not sure what the implications are.